content/shared/components/droppable.ts

```diff
@@ -0,0 +1,246 @@
+import {$html, animate, Browser, ElementView, slide, SVGParentView} from '@mathigon/boost';
+import {applyDefaults, EventTarget} from '@mathigon/core';
+import {Bounds, Point} from '@mathigon/euclid';
+
+type HoverData =
+  {tag: 'Target', $el: ElementView} |
+  {tag: 'NonTarget'};
+
+
+interface DraggableOptions {
+  $parent?: ElementView;
+  /** Whether it is draggable along the x-axis. */
+  moveX?: boolean;
+  /** Whether it is draggable along the y-axis. */
+  moveY?: boolean;
+  /** Interval for snapping (in px) */
+  snap?: number;
+  /** Whether to use CSS transforms rather than `left` and `right`. */
+  useTransform?: boolean;
+  /** Margin within the `$parent` element. */
+  margin?: number;
+  /** Custom rounding function. */
+  round?: ((p: Point) => Point);
+  /** Override `$parent` width. */
+  width?: number;
+  /** Override `$parent` height. */
+  height?: number;
+  /** Whether to reset position when dropped outside of a target */
+  resetOnMiss?: boolean;
+  /** Elements that the Draggable instance can be dropped onto */
+  $targets?: ElementView[];
+}
+
+/**
+ * A draggable and droppable HTML element.
+ * @emits Draggable#start - when the user starts dragging this element.
+ * @emits Draggable#drag {elemPos: Point, pointerPos: Point} positionData - while the user is dragging this element.
+ * @emits Draggable#click - when the user clicks on this element.
+ * @emits Draggable#end - after the user stops dragging this element.
+ * @emits Draggable#move {Point} position - When the position of this element changes.
+ * @emits Draggable#enter-target {ElementView} $target - Fires when the pointer has entered the bounds of a 'target' element while dragging
+ * @emits Draggable#exit-target {ElementView} $target - Fires when the pointer has exited the bounds of a 'target' element while dragging
+ * @emits Draggable#dropped-target {$el: ElementView, index: number} target - Fires when the user releases the pointer while over a 'target' element
+ * @emits Draggable#dropped-non-target - Fires when the user releases the pointer while not over a 'target' element
+ */
+export class Draggable extends EventTarget {
+  protected options: DraggableOptions;
+  protected areaBounds?: {topLeft: Point, bottomRight: Point};
```

Comment:
Make this a `Bounds()` object, rather than re-creating on every frame in `.setPosition()`